lib/IRGen/IRGenSIL.cpp

```diff
@@ -862,6 +862,18 @@ class IRGenSILFunction :
     return VarDecl->getInterfaceType()->hasTypeParameter();
   }
 
+  /// Force all archetypes referenced by the type to be bound by this point.
+  /// TODO: just make sure that we have a path to them that the debug info
+  ///       can follow.
+  void bindArchetypes(swift::Type Ty) {
```

Comment:
It would be nice if lldb only required type metadata for top-level generic arguments, and not any referenced associated types. Can you file a radar to fix that? It would be feasible to tackle this once you're able to use remote mirrors to calculate layouts (remote mirrors doesn't require metadata instantiation in the target).